blog/_posts/2021-03-15-scaling-inlets.md

```diff
@@ -0,0 +1,349 @@
+
+So what if we want to bring a lot of services or applications into the cluster? Are the Helm charts available to assist is us? What do you need to know to scale to tens or hundreds of inlets tunnels?
+
+Let's have a look.
+
+## A little retrospect
+
+First, let’s recap what we did in the previous post focusing on the server part of the tunnels.
+
+![prometheus](/images/2020-12-multi-cluster-monitoring/architecture.png)
+> Kubernetes multi-cluster monitoring with Prometheus, nginx ingress controller, cert-manager and inlets PRO
+
+By using the Helm chart, an inlets PRO exit server is created for each tunnel. Those tunnels' control planes are securely exposed using an ingress and a Let's Encrypt certificate created by cert-manager.
+Besides that, the data plane is only available from within the Kubernetes cluster.
+
+With this setup, every time we add another tunnel, the cert-manager will create a new certificate with Let's Encrypt for the specific subdomain.
+For a couple of services, that's ok, but if we want to scale to tens or hundreds of tunnels, perhaps this can become an issue because of Let's Encrypt's rate limits.
+
+Because all our tunnels will be using the same domain, a wildcard certificate can help us here. If you have already such a certificate available, we can put that in place and otherwise, we can let cert-manager issue a wildcard certificate with the proper DNS01 Challenge Provider.
+
+## Pre-requisites
+
+- Some Kubernetes clusters running in different locations, e.g. on a public cloud (e.g. GKE, AKS, EKS, DOKS, …) or on a Raspberry Pi in a private home-lab
+- `kubectl`, configured to connect to the cluster
+- `kubectx`, optionally, but useful to manage the different cluster
+- `arkade` - portable Kubernetes marketplace
+- A domain and access to your DNS admin panel to create a sub-domain
+
+## Preparing the cluster
+
+Install the nginx ingress controller and cert-manager with `arkade`:
+
+``` bash
+arkade install ingress-nginx --namespace ingress-nginx
+arkade install cert-manager --namespace cert-manager
+```
+
+It can take some time until the nginx pods and loadbalancer is ready, but when everything is available, add the DNS records to connect your domain. Because we will create multiple tunnel for different applications, the easiest way to go forward is to configure a wildcard DNS record (e.g. *.inlets.example.com). If you don’t like wildcard records, create a record for each application pointing to the same public IP address (e.g. grafana.inlets.example.com, keycloak.inlets.example.com, …).
+
+Next, create an Issuer, or ClusterIssuer, to use Amazon Route53 to solve DNS01 ACME challenges.
+
+> A detailed explanation on how to prepare the issuer can be found [here](https://cert-manager.io/docs/configuration/acme/dns01/route53/)
+
+``` bash
```

Comment:
Please use my example from the docs which uses an `export` to set the email address rather than relying on the person to edit the YAML after pasting it in.

https://docs.inlets.dev/#/get-started/quickstart-ingresscontroller-cert-manager?id=expose-your-ingresscontroller-and-get-tls-from-letsencrypt